HOWTO.md

```diff
@@ -187,6 +188,23 @@ Note that in practice exchanging all the attentions with a sparse alternative ma
 - on creating [complex sparsity patterns](docs/source/2d_attention_patterns.ipynb)
 - on a [SwinTransformers](docs/source/swin_transformers.ipynb)
 
+## BlockSparseAttention
+
+This uses the blocksparse computations which are provided by Triton out of the box. In xFormers, you can use the BlocksSparseAttention,
+which will limit the computations and the memory to the tiles specified in the `layout`:
+
+```python
+import torch
+from xformers.components.attention import BlockSparseAttention
+BLOCK_SIZE = 32     # for instance
+SEQ = 1024          # ..
+layout = torch.tril(torch.ones(SEQ//BLOCK_SIZE, SEQ//BLOCK_SIZE)).cuda()  # Only compute the lower triangular
+lower_tril_attention = BlockSparseAttention(layout=layout, block_size=BLOCK_SIZE, dropout=0.1)
+causal_mask = torch.tril(torch.ones(SEQ, SEQ)).bool().cuda()
+...
+att = lower_tril_attention(k, q, v, att_mask=causal_mask)
```

Comment:
I think that this could do with a second and third pass :) Would you mind adding that in another PR, re-reading some of this with an NLP perspective for instance ? I must have had quite a few blind spots